pkg/ccl/changefeedccl/sink_cloudstorage.go

```diff
@@ -187,10 +190,12 @@ func (s *cloudStorageSink) EmitRow(
 		file = &cloudStorageSinkFile{}
 		s.files[key] = file
 	}
-	if file.earliestTs.IsEmpty() || updated.Less(file.earliestTs) {
-		file.earliestTs = updated
-	}
 
+	// Keep track of the time of the last call to this method before a Flush.
+	// Using this timestamp in the output file names allows us to ensure that,
+	// during consumption, timestamps are consumed in the same order that they're
+	// emitted in.
+	file.lastUpdatedTs = s.clock.Now()
```

Comment:
Is there a way we can make this approach work without needing to read the clock for every single row? It seems like there should be.

Reply:
This is doable. See following comment.